### DISCUSSION TITLE
[3.1] Many server middlewares issue ?

### DISCUSSION THREAD

**Discussion by Loyz07** [nuxt/nuxt#18515] [Questions]
Hello,

I've just updated nuxt to v3.1 and since this update I don't manage to run many server middlewares anymore.

I have 2 server middlewares `logger.ts` and `authentication.ts`. Only `logger.ts` is executed on requests.
If I add a 3rd middleware `test.ts` only this one is executed.
If I rename `test.ts` to `a.ts` only `logger.ts` is executed.

It seems that only the last middleware (A-Z sorted) is executed.

Someone could confirm so that I can submit an issue ?

Thanks.

**Comment by ndragun92** [comment]
Could you also make a small reproduction on StackBlitz or somewhere?

**Comment by Loyz07** [comment]
Sure.

Here is a link:
https://stackblitz.com/edit/nuxt-starter-511bme?file=server/middleware/a.ts

`a` is not console.logged.

**Reply by ndragun92** [reply]
Thanks yea also to me it looks like a bug.... Nice spot

**Comment by yassilah** [comment]
Ok I think it is linked to one of my PRs on Nitro: https://github.com/unjs/nitro/pull/807 It was intended to filter unique handlers per route/method but middlewares all have the same route I think; I'll make a quick PR to revert that
